akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1013,11 +1012,7 @@ trait FlowOps[+Out, +Mat] {
    * '''Cancels when''' downstream cancels
    *
    */
-  def flatten[U](strategy: FlattenStrategy[Out, U]): Repr[U, Mat] = strategy match {
-    case scaladsl.FlattenStrategy.Concat | javadsl.FlattenStrategy.Concat ⇒ deprecatedAndThen(ConcatAll())
-    case _ ⇒
-      throw new IllegalArgumentException(s"Unsupported flattening strategy [${strategy.getClass.getName}]")
-  }
+  def flattenConcat[U]()(implicit ev: Out <:< Source[U, _]): Repr[U, Mat] = deprecatedAndThen(ConcatAll())
```

Comment:
😊

the reason I didn't call it flatConcat is because it doesn't take a Out =>
Source[U, _] function.

But I am starting to wonder whether flatConcat wouldn't be a better
feature, since it wouldn't need a blind type ascription in the Java DSL and
a witness in the Scala DSL.

flattenConcat would then be isomorphic to flatConcat(identity)

-- 
Cheers,
√
On 3 Nov 2015 11:28, "Konrad Malawski" <notifications@github.com> wrote:

> In akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala
> <https://github.com/akka/akka/pull/18809#discussion_r43733137>:
>
> > @@ -1013,11 +1012,7 @@ trait FlowOps[+Out, +Mat] {
> >     * '''Cancels when''' downstream cancels
> >     *
> >     */
> > -  def flatten[U](strategy: FlattenStrategy[Out, U]): Repr[U, Mat] = strategy match {
> > -    case scaladsl.FlattenStrategy.Concat | javadsl.FlattenStrategy.Concat ⇒ deprecatedAndThen(ConcatAll())
> > -    case _ ⇒
> > -      throw new IllegalArgumentException(s"Unsupported flattening strategy [${strategy.getClass.getName}]")
> > -  }
> > +  def flattenConcat[U]()(implicit ev: Out <:< Source[U, _]): Repr[U, Mat] = deprecatedAndThen(ConcatAll())
>
> <https://imgflip.com/i/tjomq>
>
> Good to stay consistent between DSLs [image: :+1:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/akka/akka/pull/18809/files#r43733137>.
>